servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/MutualSslTest.java

```diff
@@ -44,32 +56,57 @@
     public final Timeout timeout = new ServiceTalkTestTimeout();
     private final SslProvider serverSslProvider;
     private final SslProvider clientSslProvider;
+    @SuppressWarnings("rawtypes")
+    private final Map<SocketOption, Object> serverListenOptions;
+    @SuppressWarnings("rawtypes")
+    private final Map<SocketOption, Object> clientOptions;
 
-    public MutualSslTest(final SslProvider serverSslProvider, final SslProvider clientSslProvider) {
+    public MutualSslTest(final SslProvider serverSslProvider, final SslProvider clientSslProvider,
+                         @SuppressWarnings("rawtypes") final Map<SocketOption, Object> serverListenOptions,
+                         @SuppressWarnings("rawtypes") final Map<SocketOption, Object> clientOptions) {
         this.serverSslProvider = serverSslProvider;
         this.clientSslProvider = clientSslProvider;
+        this.serverListenOptions = serverListenOptions;
+        this.clientOptions = clientOptions;
     }
 
-    @Parameterized.Parameters(name = "server={0} client={1}")
+    @Parameterized.Parameters(name = "{index}: server={0} client={1} server opts={2} client opts={3}")
     public static Collection<Object[]> sslProviders() {
-        return asList(
-                new Object[]{JDK, JDK},
-                new Object[]{JDK, OPENSSL},
-                new Object[]{OPENSSL, JDK},
-                new Object[]{OPENSSL, OPENSSL}
-        );
+        final SslProvider[] providers = new SslProvider[] {JDK, OPENSSL};
+        @SuppressWarnings("rawtypes")
+        final List<Map<SocketOption, Object>> serverOpts = asList(emptyMap(), serverTcpFastOpenOptions());
+        @SuppressWarnings("rawtypes")
+        final List<Map<SocketOption, Object>> clientOpts = asList(emptyMap(), clientTcpFastOpenOptions());
+        final List<Object[]> results = new ArrayList<>(
+                providers.length * 2 * serverOpts.size() * clientOpts.size());
+        for (SslProvider serverProvider : providers) {
```

Comment:
This is pretty ugly, you could probably replace this with junit's `@Theory` & `@Datapoints`

Reply:
thx for suggesting! done.